src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,951 @@
+        /// }
+        /// 
+        /// ---------------------------------------------------->
+        /// 
+        /// public int GetF(int x, int y)
+        /// {
+        ///     return x * y;
+        /// }
+        /// 
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = x * y;
+        /// }
+        /// </summary>
+        private static async Task<SyntaxNode> ExtractMethodAsync(Document document, TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, ImmutableArray<IParameterSymbol> validParameters, string newMethodIdentifier, SyntaxGenerator generator, CancellationToken cancellationToken)
+        {
+            // Remove trivia so the expression is in a single line and does not affect the spacing of the following line
+            var newStatements = ImmutableArray.CreateBuilder<SyntaxNode>();
+            newStatements.Add(generator.ReturnStatement(expression.WithoutTrivia()));
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var typeSymbol = semanticModel.GetTypeInfo(expression, cancellationToken).ConvertedType ?? semanticModel.Compilation.ObjectType;
+            var codeGenerationService = document.GetRequiredLanguageService<ICodeGenerationService>();
+            var newMethod = CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, name: newMethodIdentifier, parameters: validParameters, statements: newStatements.ToImmutable(), returnType: typeSymbol);
+            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var newMethodDeclaration = codeGenerationService.CreateMethodDeclaration(newMethod, options: new CodeGenerationOptions(options: options, parseOptions: expression.SyntaxTree.Options));
+            return newMethodDeclaration;
+        }
+
+        /// <summary>
+        /// Generates a method declaration containing a call to the method that introduced the parameter.
+        /// Example:
+        /// 
+        /// ***This is an intermediary step in which the original function has not be updated yet
+        /// public void M(int x, int y)
+        /// {
+        ///     int f = [|x * y|];
+        /// }
+        /// 
+        /// ---------------------------------------------------->
+        /// 
+        /// public void M(int x, int y) // Generated overload
+        /// {
+        ///     M(x, y, x * y);
+        /// }
+        /// 
+        /// public void M(int x, int y) // Original function
+        /// {
+        ///     int f = x * y;
+        /// }
+        /// </summary>
+        private static async Task<SyntaxNode> GenerateNewMethodOverloadAsync(Document document, TExpressionSyntax expression,
+            IMethodSymbol methodSymbol, int insertionIndex, SyntaxGenerator generator, CancellationToken cancellationToken)
+        {
+            var arguments = generator.CreateArguments(methodSymbol.Parameters);
```

Comment:
this is interesting... and i'm not srue i understand.... you're making an argument list... out of parameter symbols?  can you doc what this is for?

Reply:
For the overload case I need to create an invocation of the original method. For that I need all the parameters as arguments